src/main/java/pl/betoncraft/betonquest/conditions/PartialDateCondition.java

```diff
@@ -0,0 +1,136 @@
+package pl.betoncraft.betonquest.conditions;
+
+import com.google.common.base.Joiner;
+import pl.betoncraft.betonquest.Instruction;
+import pl.betoncraft.betonquest.InstructionParseException;
+import pl.betoncraft.betonquest.QuestRuntimeException;
+import pl.betoncraft.betonquest.api.Condition;
+
+import java.util.Calendar;
+import java.util.HashSet;
+import java.util.Set;
+
+
+/**
+ * Checks if the current date matches one of the given dates
+ *
+ * @author Jonas Blocher
+ */
+public class PartialDateCondition extends Condition {
+
+    private final Set<Integer> dayOfMonth;
+    private final Set<Integer> month;
+    private final Set<Integer> year;
+
+    public PartialDateCondition(Instruction instruction) throws InstructionParseException {
+        super(instruction);
+        super.staticness = true;
+        super.persistent = true;
+        dayOfMonth = new HashSet<>();
+        month = new HashSet<>();
+        year = new HashSet<>();
+        String dayOfMonthString = instruction.getOptional("day");
+        if (dayOfMonthString != null) {
+            if (dayOfMonthString.matches("[^-,]*")) {
+                int i = instruction.getInt(dayOfMonthString, -1);
+                if (i < 1 || i > 31)
+                    throw new InstructionParseException(dayOfMonthString + " is not a valid day");
+                dayOfMonth.add(i);
+            } else {
+                if (!dayOfMonthString.matches("[0-9]{1,2}(-[0-9]{1,2})?(,[0-9]{1,2}(-[0-9]{1,2})?)*"))
+                    throw new InstructionParseException("could not parse days from " + dayOfMonthString);
+                String[] args1 = dayOfMonthString.split(",");
+                for (String arg : args1) {
+                    if (arg.contains("-")) {
+                        int index = arg.indexOf("-");
+                        int beginning = instruction.getInt((arg.substring(0,index)), -1);
+                        int end = instruction.getInt(arg.substring(index + 1), -1);
+                        if (beginning >= end) throw new InstructionParseException("day " + beginning + " is before " + end);
+                        for (int i = beginning; i <= end; i++) {
+                            if (i < 1 || i > 31 ) throw new InstructionParseException(dayOfMonthString + " contains invalid days");
+                            dayOfMonth.add(i);
+                        }
+                    } else {
+                        int i = instruction.getInt(arg, -1);
+                        if (i < 1 || i > 31 ) throw new InstructionParseException(dayOfMonthString + " contains invalid days");
+                        dayOfMonth.add(i);
+                    }
+                }
+            }
+        }
+        String monthString = instruction.getOptional("month");
+        if (monthString != null) {
```

Comment:
Lots of duplicated code. It works, but it's not clean. If you don't want to change it then it's fine. I'll just redirect eventual bug reports to you 😉 

In case you'd like to change it, here's my suggestion, the way I see it:
Create a `Range` class. It will take `min` and `max` values when created, and then will have `boolean checkNumber(int number)` method which will test if a number is in range. Then store those `Range`s in those maps above (`dayOfMonth`, `month`, `year`). Note that a single day could be represented as a `Range` with `min == max`.
You could check those ranges in the `check()` method with something like this:

```java
year.stream().anyMatch(range -> range.checkNumber(number));
```

Creating `Range`s could be easy - since each range is separated with a comma, you can split the string and iterate over those substrings, creating new `Range`s. Then, in the constructor, either parse the two numbers separated with dash or simply double a single number to both `min` and `max` fields. Additionally specify in the constructor lower and upper boundaries (like 1 and 31, or 1 and 12) and check the parsed `min` and `max` fields against them.

This way you have a nice, reusable `Range` which can be applied to all three time types (and more, you could even use it for hours, minutes or seconds; even weeks and days of weeks could work that way).

Reply:
Yes IntellIJ also told me I should remove this duplicated code when I was working on this but back then I was to lazy. I'll try to have a look into it tomorrow evening.